From 33281840fff2036074accfb97bf3884293960d7d Mon Sep 17 00:00:00 2001
From: "Bobby R. Bruce" <bbruce@ucdavis.edu>
Date: Tue, 19 Apr 2022 11:28:29 -0700
Subject: [PATCH 582/757] tests,gpu-compute: Replace gcn3 tests with vega
 testlib tests

Any X86 quick tests were set to use GCN3_86. This was to minimize the
number of gem5 targets to build during the quick tests. Since the
introduction of VEGA, we wish to test VEGA during the quick tests. As
such, this patch will replace all GCN3_X86 tests with VEGA_X86 tests.

Change-Id: Ibdd7832e43a284936db5102d41161388bc721bd1
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/59012
Reviewed-by: Matthew Poremba <matthew.poremba@amd.com>
Reviewed-by: Matt Sinclair <mattdsinclair@gmail.com>
Maintainer: Matt Sinclair <mattdsinclair@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>
---
 tests/gem5/cpu_tests/test.py                       | 4 ++--
 tests/gem5/hello_se/test_hello_se.py               | 8 ++++----
 tests/gem5/kvm-fork-tests/test_kvm_fork_run.py     | 2 +-
 tests/gem5/kvm-switch-tests/test_kvm_cpu_switch.py | 2 +-
 tests/gem5/learning_gem5/part1_test.py             | 4 ++--
 tests/gem5/learning_gem5/part2_test.py             | 4 ++--
 tests/gem5/m5_util/test_exit.py                    | 2 +-
 tests/gem5/test_build/test_build.py                | 2 +-
 tests/gem5/x86-boot-tests/test_linux_boot.py       | 2 +-
 9 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/tests/gem5/cpu_tests/test.py b/tests/gem5/cpu_tests/test.py
index a962337..c0322b2 100644
--- a/tests/gem5/cpu_tests/test.py
+++ b/tests/gem5/cpu_tests/test.py
@@ -46,7 +46,7 @@ from testlib import *
 workloads = ('Bubblesort','FloatMM')
 
 valid_isas = {
-    constants.gcn3_x86_tag :
+    constants.vega_x86_tag :
         ('AtomicSimpleCPU', 'TimingSimpleCPU', 'DerivO3CPU'),
     constants.arm_tag:
         ('AtomicSimpleCPU', 'TimingSimpleCPU', 'MinorCPU', 'DerivO3CPU'),
@@ -60,7 +60,7 @@ base_path = joinpath(config.bin_path, 'cpu_tests')
 base_url = config.resource_url + '/test-progs/cpu-tests/bin/'
 
 isa_url = {
-    constants.gcn3_x86_tag : base_url + "x86",
+    constants.vega_x86_tag : base_url + "x86",
     constants.arm_tag : base_url + "arm",
     constants.riscv_tag : base_url + "riscv",
 }
diff --git a/tests/gem5/hello_se/test_hello_se.py b/tests/gem5/hello_se/test_hello_se.py
index e1e6aaf..9cbfa52 100644
--- a/tests/gem5/hello_se/test_hello_se.py
+++ b/tests/gem5/hello_se/test_hello_se.py
@@ -55,7 +55,7 @@ isa_str_map = {
 }
 
 static_progs = {
-    constants.gcn3_x86_tag: (
+    constants.vega_x86_tag: (
         "x86-hello64-static",
         "x86-hello32-static",
     ),
@@ -68,10 +68,10 @@ static_progs = {
     constants.sparc_tag: ("sparc-hello",),
 }
 
-dynamic_progs = {constants.gcn3_x86_tag: ("x86-hello64-dynamic",)}
+dynamic_progs = {constants.vega_x86_tag: ("x86-hello64-dynamic",)}
 
 cpu_types = {
-    constants.gcn3_x86_tag: ("timing", "atomic", "o3"),
+    constants.vega_x86_tag: ("timing", "atomic", "o3"),
     constants.arm_tag: ("timing", "atomic", "o3", "minor"),
     constants.mips_tag: ("timing", "atomic", "o3"),
     constants.riscv_tag: ("timing", "atomic", "o3", "minor"),
@@ -81,7 +81,7 @@ cpu_types = {
 # We only want to test x86, arm, and riscv on quick. Mips and sparc will be
 # left for long.
 os_length = {
-    constants.gcn3_x86_tag: constants.quick_tag,
+    constants.vega_x86_tag: constants.quick_tag,
     constants.arm_tag: constants.quick_tag,
     constants.mips_tag: constants.long_tag,
     constants.riscv_tag: constants.quick_tag,
diff --git a/tests/gem5/kvm-fork-tests/test_kvm_fork_run.py b/tests/gem5/kvm-fork-tests/test_kvm_fork_run.py
index 0ddc350..b7986c7 100644
--- a/tests/gem5/kvm-fork-tests/test_kvm_fork_run.py
+++ b/tests/gem5/kvm-fork-tests/test_kvm_fork_run.py
@@ -59,7 +59,7 @@ def test_kvm_fork_run(cpu: str, num_cpus: int, mem_system: str, length: str):
         isa_to_use = constants.x86_tag
     else:
         protocol_to_use = None
-        isa_to_use = constants.gcn3_x86_tag
+        isa_to_use = constants.vega_x86_tag
 
     gem5_verify_config(
         name=name,
diff --git a/tests/gem5/kvm-switch-tests/test_kvm_cpu_switch.py b/tests/gem5/kvm-switch-tests/test_kvm_cpu_switch.py
index 4d501e4..ede90db 100644
--- a/tests/gem5/kvm-switch-tests/test_kvm_cpu_switch.py
+++ b/tests/gem5/kvm-switch-tests/test_kvm_cpu_switch.py
@@ -59,7 +59,7 @@ def test_kvm_switch(cpu: str, num_cpus: int, mem_system: str, length: str):
         isa_to_use = constants.x86_tag
     else:
         protocol_to_use = None
-        isa_to_use = constants.gcn3_x86_tag
+        isa_to_use = constants.vega_x86_tag
 
     gem5_verify_config(
         name=name,
diff --git a/tests/gem5/learning_gem5/part1_test.py b/tests/gem5/learning_gem5/part1_test.py
index 4153165..32dc666 100644
--- a/tests/gem5/learning_gem5/part1_test.py
+++ b/tests/gem5/learning_gem5/part1_test.py
@@ -36,7 +36,7 @@ gem5_verify_config(
     config_args = [],
     length = constants.quick_tag,
     valid_isas=(
-        constants.gcn3_x86_tag,
+        constants.vega_x86_tag,
         constants.riscv_tag,
         constants.arm_tag,
     ),
@@ -50,7 +50,7 @@ gem5_verify_config(
     config_args = [],
     length = constants.quick_tag,
     valid_isas=(
-        constants.gcn3_x86_tag,
+        constants.vega_x86_tag,
         constants.riscv_tag,
         constants.arm_tag
     ),
diff --git a/tests/gem5/learning_gem5/part2_test.py b/tests/gem5/learning_gem5/part2_test.py
index 24d623c..f3658b7 100644
--- a/tests/gem5/learning_gem5/part2_test.py
+++ b/tests/gem5/learning_gem5/part2_test.py
@@ -52,7 +52,7 @@ gem5_verify_config(
     config=joinpath(config_path, 'simple_memobj.py'),
     config_args = [],
     # note: by default the above script uses x86
-    valid_isas=(constants.gcn3_x86_tag,),
+    valid_isas=(constants.vega_x86_tag,),
 )
 
 gem5_verify_config(
@@ -61,7 +61,7 @@ gem5_verify_config(
     config=joinpath(config_path, 'simple_cache.py'),
     config_args = [],
     # note: by default the above script uses x86
-    valid_isas=(constants.gcn3_x86_tag,),
+    valid_isas=(constants.vega_x86_tag,),
 )
 
 # Note: for simple memobj and simple cache I want to use the traffic generator
diff --git a/tests/gem5/m5_util/test_exit.py b/tests/gem5/m5_util/test_exit.py
index d6695df..d06e703 100644
--- a/tests/gem5/m5_util/test_exit.py
+++ b/tests/gem5/m5_util/test_exit.py
@@ -70,5 +70,5 @@ gem5_verify_config(
         resource_path,
         "x86",
     ],
-    valid_isas=(constants.gcn3_x86_tag,),
+    valid_isas=(constants.vega_x86_tag,),
 )
diff --git a/tests/gem5/test_build/test_build.py b/tests/gem5/test_build/test_build.py
index 715b129..5be5d5d 100644
--- a/tests/gem5/test_build/test_build.py
+++ b/tests/gem5/test_build/test_build.py
@@ -31,7 +31,7 @@ import re
 import os
 from testlib import *
 
-common_isas = [constants.gcn3_x86_tag, constants.arm_tag, constants.riscv_tag]
+common_isas = [constants.vega_x86_tag, constants.arm_tag, constants.riscv_tag]
 skipped_isas = {constants.null_tag, constants.all_compiled_tag}
 
 for isa in constants.supported_isas:
diff --git a/tests/gem5/x86-boot-tests/test_linux_boot.py b/tests/gem5/x86-boot-tests/test_linux_boot.py
index 7e3ad4e..1cea4d8 100644
--- a/tests/gem5/x86-boot-tests/test_linux_boot.py
+++ b/tests/gem5/x86-boot-tests/test_linux_boot.py
@@ -70,7 +70,7 @@ def test_boot(
         isa_to_use=constants.x86_tag
     else:
         protocol_to_use=None
-        isa_to_use=constants.gcn3_x86_tag
+        isa_to_use=constants.vega_x86_tag
 
     gem5_verify_config(
         name=name,
-- 
1.8.3.1

